### DISCUSSION TITLE
Parse JSDoc comments on Properties

### DISCUSSION THREAD

**Discussion by dudeofawesome** [mikro-orm/mikro-orm#5313] [Ideas]
**Is your feature request related to a problem? Please describe.**
When defining an Entity, I often find myself writing JSDoc code comments to provide inline documentation for properties. I usually want those comments to also exist on the database representation of the Entity as well, so I have to copy my comments into the `comment` property of the `Property` decorator.

**Describe the solution you'd like**
It would be nice if we could parse the code comments from the AST just like we parse the property name and type.

[Here's an example of TS-Morph providing the JSDoc comment on `bar` into the `Node::jsDoc[0].comment` property.](https://ts-ast-viewer.com/#code/N4KABBYPQFQ2AxA9ksBjANgQwM47ACZJpgxTiSa77KqiQPRwWMTwAqAFgJb69gBbAJ7okAgQFMAdgBdCEnGgBO3AEbcpAczAB3TljmqsSwkgUA6Fq3gBJOWixSwEgG7SwqiYICuGGdwwNCQBCK0YyMMgjJQAuMBwZFS0wAF4wAHIjAC90gG4rAF8QAqA)

**Comment by B4nan** [comment]
I am personally not interested in implementing that, but the metadata provider layer is (and always was) configurable, nothing holds you from creating such metadata provider, and maybe publish it as a 3rd party package.

**Reply by dudeofawesome** [reply]
Does that mean that Mikro ORM would not be open to a PR implementing this?

**Reply by B4nan** [reply]
Oh wait a second, I think I misunderstood what you are asking for. I thought you wanted a separate metadata provider that works based on JSDoc, but you want to improve the existing one from `@mikro-orm/reflection` so it can read property comments from JSDoc? 

In such case, yes, PR welcome. I am not entirely sure if it's a good idea to do this all the time, but it could be conditional.
